python_modules/libraries/dagster-components/dagster_components/core/schema/objects.py

```diff
@@ -1,94 +1,62 @@
 from collections.abc import Mapping, Sequence
-from typing import Annotated, Any, Callable, Literal, Optional, Union
+from typing import Any, Callable, Literal, Optional, Union
 
 import dagster._check as check
+from dagster._core.definitions.asset_dep import AssetDep
 from dagster._core.definitions.asset_key import AssetKey
 from dagster._core.definitions.asset_selection import AssetSelection
 from dagster._core.definitions.asset_spec import AssetSpec, map_asset_specs
 from dagster._core.definitions.assets import AssetsDefinition
-from dagster._core.definitions.declarative_automation.automation_condition import (
-    AutomationCondition,
-)
 from dagster._core.definitions.definitions_class import Definitions
 from dagster._record import replace
 from pydantic import BaseModel
 
-from dagster_components.core.schema.base import ResolvableModel
-from dagster_components.core.schema.metadata import ResolvableFieldInfo
-from dagster_components.core.schema.resolver import TemplatedValueResolver
+from dagster_components.core.schema.base import ResolvableModel, Resolver, resolver
+from dagster_components.core.schema.context import ResolutionContext
 
 
-class OpSpecModel(ResolvableModel["OpSpecModel"]):
+class OpSpecModel(ResolvableModel):
     name: Optional[str] = None
     tags: Optional[dict[str, str]] = None
 
-    def resolve(self, resolver: TemplatedValueResolver) -> "OpSpecModel":
-        return OpSpecModel(**resolver.resolve_obj(self.model_dump(exclude_unset=True)))
+
+class AssetDepModel(ResolvableModel[AssetDep]):
```

Comment:
in terms of naming convention given where this landed we should consider `Schema` instead of `Model` IMO

Reply:
hm I think the fact that we inherit from `BaseModel` makes me lean towards `ResolvableModel` to make the connection more obvious. Not super strongly held though